baseclasses/problems/pyAero_problem.py

```diff
@@ -352,7 +352,15 @@ def __init__(self, name, **kwargs):
 
         self.possibleDVs = set()
         for var in self.allVarFuncs:
-            if getattr(self, var) is not None:
+            validDV = False
+            # Flow state variables are only valid DVs if they were specified in the constructor
+            if var in self.fullState:
+                if var in self.inputs:
+                    validDV = True
+            else:
+                if getattr(self, var) is not None:
+                    validDV = True
+            if validDV:
```

Comment:
Not sure that'd be a good idea because that set would only be used here. But I cleaned up the logic a bit

Reply:
Sorry for the delay on this. Just to comment quickly, yes, its probably not used elsewhere. However, I think it is a bit more maintainable since the sets should not intersect, and then you need to update both list. This would be something like 
```
notStateVars = {
    "alpha",
    "areaRef",
    "beta",
    "chordRef",
    "momentAxis",
    "phat",
    "qhat",
    "rhat",
    "spanRef",
    "xRef",
    "xRot",
    "yRef",
    "yRot",
    "zRef",
    "zRot",
}

allVarFuncs = notStateVars + fullState
```
so there is no need to have `allVarFuncs`. Does this make sense? I might be missing something.

Reply:
I think the issue is that not everything in `fullState` can be a DV (for example `mu` and `nu`). So we'll always need to have 2 lists, however we arrange it, in order to define both all of the state variables and all of the state variables that can be used as design variables